src/main/java/de/allcom/services/ProductService.java

```diff
@@ -0,0 +1,137 @@
+package de.allcom.services;
+
+import de.allcom.dto.product.CreateProductRequestDto;
+import de.allcom.dto.product.ProductDto;
+import de.allcom.dto.product.UpdateProductRequestDto;
+import de.allcom.exceptions.CopyPhotoException;
+import de.allcom.exceptions.NotFoundException;
+import de.allcom.models.Category;
+import de.allcom.models.Product;
+import de.allcom.models.PhotoProduct;
+import de.allcom.repositories.CategoryRepository;
+import de.allcom.repositories.PhotoProductRepository;
+import de.allcom.repositories.ProductRepository;
+import de.allcom.services.utils.Converters;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+    @Value("${photo.upload.path}")
+    private String uploadPath;
+
+    private final ProductRepository productRepository;
+    private final PhotoProductRepository photoProductRepository;
+    private final CategoryRepository categoryRepository;
+    private final Converters converters;
+
+    public Page<ProductDto> getAllProducts(PageRequest pageRequest) {
+        Page<Product> products = productRepository.findAll(pageRequest);
+        return products.map(converters::fromProductToProductDto);
+    }
+
+    @Transactional
+    public ProductDto createProductWithPhotos(CreateProductRequestDto request) {
+        Product product = new Product();
+        product.setName(request.getName());
+        product.setDescription(request.getDescription());
+        product.setWeight(request.getWeight());
+        product.setCategory(request.getCategory());
+        Product savedProduct = productRepository.save(product);
+        if (request.getPhotos() != null) {
+            List<PhotoProduct> photos = uploadPhotos(request.getPhotos(), savedProduct);
+            savedProduct.setPhotos(photos);
+        } else {
+            throw new IllegalArgumentException("Null original file name");
+        }
+        productRepository.save(savedProduct);
```

Comment:
Can we use one time `productRepository.save()` method?

Reply:
ne